app/assets/javascripts/discourse/tests/acceptance/user-private-messages-test.js

```diff
@@ -153,6 +159,224 @@ acceptance(
+        payload: {
+          last_read_post_number: null,
+          highest_post_number: 1,
+          group_ids: group_ids || [],
+        },
+      });
+    };
+
+    const publishArchiveToMessageBus = function () {
+      publishToMessageBus("/private-message-topic-tracking-state/5", {
+        topic_id: Math.random(),
+        message_type: "archive",
+      });
+    };
+
+    const publishGroupArchiveToMessageBus = function (group_ids) {
+      publishToMessageBus("/private-message-topic-tracking-state/5", {
+        topic_id: Math.random(),
+        message_type: "group_archive",
+        payload: {
+          group_ids: group_ids,
+        },
+      });
+    };
+
+    test("incoming archive message on all and archive filter", async function (assert) {
+      for (const url of [
+        "/u/charlie/messages",
+        "/u/charlie/messages/archive",
+        "/u/charlie/messages/personal",
+        "/u/charlie/messages/personal/archive",
+      ]) {
+        await visit(url);
+
+        publishArchiveToMessageBus();
+
+        await visit(url); // wait for re-render
+
+        assert.ok(
+          exists(".show-mores"),
+          `${url} displays the topic incoming info`
+        );
+      }
+
+      for (const url of [
+        "/u/charlie/messages/group/awesome_group/archive",
+        "/u/charlie/messages/group/awesome_group",
+      ]) {
+        await visit(url);
+
+        publishArchiveToMessageBus();
+
+        await visit(url); // wait for re-render
+
+        assert.ok(
+          !exists(".show-mores"),
+          `${url} does not display the topic incoming info`
+        );
+      }
+    });
+
+    test("incoming group archive message on all and archive filter", async function (assert) {
+      for (const url of [
+        "/u/charlie/messages",
+        "/u/charlie/messages/archive",
+        "/u/charlie/messages/group/awesome_group",
+        "/u/charlie/messages/group/awesome_group/archive",
+      ]) {
+        await visit(url);
+
+        publishGroupArchiveToMessageBus([14]);
+
+        await visit(url); // wait for re-render
+
+        assert.ok(
+          exists(".show-mores"),
+          `${url} displays the topic incoming info`
+        );
+      }
+
+      for (const url of [
+        "/u/charlie/messages/personal",
+        "/u/charlie/messages/personal/archive",
+      ]) {
+        await visit(url);
+
+        publishGroupArchiveToMessageBus([14]);
+
+        await visit(url); // wait for re-render
+
+        assert.ok(
+          !exists(".show-mores"),
+          `${url} does not display the topic incoming info`
+        );
+      }
+    });
+
+    test("incoming unread and new messages on all filter", async function (assert) {
+      await visit("/u/charlie/messages");
+
+      publishUnreadToMessageBus();
+      publishNewToMessageBus();
+
+      await visit("/u/charlie/messages"); // wait for re-render
+
+      assert.equal(
+        queryAll(".messages-nav li a.new").text().trim(),
```

Comment:
Didnt look at code, but all these queryAll here seem weird, is there multiple li a.new ? Also text is jquery function we probably should avoid using this.

```
document.querySelectorAll("div").text() // Uncaught TypeError: document.querySelectorAll(...).text is not a function
    at <anonymous>:1:34
```

Reply:
Good point. I switched it to use `query` and `innerText`